content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,556 @@
+    
+### Opinion 
+
+The node updates its local opinion on the consensus proposal by
+examining the relationship between the evidence accumulated for a
+proposal with the confidence encoded in the `alpha` parameter:
+
+    IF
+      evidence > alpha
+    THEN ;; The node adopts the opinion YES on the proposal
+      opinion <-- YES
+    ELSE IF       
+      evidence < 1 - alpha
+    THEN;; The node adopts the opinion NO on the proposal 
+      opinion <-- NO
+       
+If the opinion of the node is `UNDECIDED` after evaluating the opinion
+phase, the number of uniform randomly queries nodes is adjusted by
+multiplying the neighbors `k` by the `k_multiplier` up to the limit
+of `k_max_multiplier` query size increases.
+    
+    ;; possibly increase number nodes to uniformly randomly query in next round
+    WHEN
+       opinion == UNDECIDED
+    AND 
+       k < k_original * k_multiplier ^ max_k_multiplier
+    THEN 
+       k <-- k * k_multiplier
+
+###  Decision 
+
+After the OPINION phase is executed, the current value of `confidence`
+is considered: if `confidence` exceeds a threshold derived from the
+network size and directly related to the total votes received, an
+honest node marks the decision as final, and always returns this
+opinion is response to further queries from other nodes on the
+network.
+ 
+    IF 
+      confidence > confidence_threshold
+    OR 
+      round > max_rounds
+    THEN
+      finalized <-- T
+      QUERY LOOP TERMINATES
+    ELSE 
+      round += 1
+      GOTO QUERY 
+
+Thus, after the decision phase, either a decision has been finalized
+and the local node becomes quiescent never initiating a new query, or
+it initiates a new query.
+
+## Termination Conditions
+
+The alogrithim
+
+## Further points
+
+### Node receives information when answering queries
+
+In the query step, the node is envisioned as packing information into
+the query to cut down on the communication overhead a query to each of
+this $k$ nodes containing the node's own current opinion on the
+proposal ("YES", "NO", or "UNDECIDED").  The relation of the metadata
+is not currently analyzed to these opinions should be constrained.
+
+### Weighted Node values
+
+The view of network peers participants may optionally have a weighting
+assigned for each node consisting of a real number on the interval $[0, 1]$.
+This weight is used in each query round when selecting the $k$
+peers so the probability of selecting nodes is proportional to their
+weight.
+
+$$
+P(i) = \frac{w_i}{\sum_{j=0}^{j=N} w_j}
+$$ 
+
+where $w_i$ is the weight of the {i}th peer.
+
+This weighted value would be used to reflect decisions mediating
+
+- Staking
+- Heuristic reputation
+- Manual reputation
+
+### Problems 
+
+If the view of other nodes is incomplete, then the sum of the optional
+weighting won't be a probability distribution normalized to 1.
+
+The current algorithm doesn't describe how the initial opinions are formed.
```

Comment:
By default an honest node would vote `yes` for anything that is not in conflict with previous voted things (and finalised ones). Otherwise is up the heuristics of resolving conflicts. But, for a happy path, every honest nodes always votes on `yes`.